### ISSUE TITLE
Get snake game input working

### ISSUE THREAD

**Issue by hintron** [hintron/computer-enhance#49]
"Input is handled via the input "register" at memory address 0xFFFE, with each bit corresponding to an input button. Starting from the low bit the buttons are: Up, Down, Left, Right, Menu. Ie the low bit is used for the up button, the 5th bit is used for the menu button, and the top 3 bits are unused."

```
| 7 | 6 | 5 | 4 | 3 | 2 | 1 | 0 |
---------------------------------
| x | x | x | M | R | L | D | U |
```

**Comment by hintron** [comment]
I think I will map `wasd` to Up, Left, Down, Right, respectively. I will make the menu button be `m`.

**Comment by hintron** [comment]
I got this hooked up, but I am struggling to get predictable and usable control of the snake. Sometimes I can get it to move around, other times, it won't respond. I inspected the log output to see what instructions I'm running, and it seems that I am loading in values into the control register at 0xFFFE, and the program is successfully detecting that I pressed up and it's loading Direction_up into the direction address. But for whatever reason, I don't see the snake move. 

I think I need to run snake in my debugger and step through instruction by instruction to see what is going on.

**Comment by hintron** [comment]
Not sure why, but I am able to get the snake moving more predictably now on Linux (with a higher framerate). But my up and down movements are inverted, and this puzzled me for a while because I could walk through the code and see that I didn't mix up the up and down, and that the execution of the code was correctly getting an up or down press according to the assembly code. 

I got the title screen image to display, and it revealed why my up and down is inverted - the entire image is inverted! The title screen prints out, but it is upside down.

![image](https://github.com/hintron/computer-enhance/assets/9166578/d9e8e0c8-2502-4610-850f-1e75b33ff6ad)

Another issue I'm seeing is that sometimes my commands are not real-time. When the FPS is super high, it seems that the commands get backed up and are delayed. When the FPS is low, it seems that the commands get missed entirely.

Also, the FPS is super high in the debugger, but low on the command line. And on the command line, using System Monitor, I can see that my program grows in memory until it exhausts system memory. That is what causes the "Killed" message - Linux is killing the process for taking too much memory.

**Comment by hintron** [comment]
Initial attempt implemented in commits 924e743...d043b69.

There are still issues to figure out with the input working, but I will resolve that in #53
